### PR TITLE
push_data_ready_event() deadlock (issue-22)

### PR DESCRIPTION
**Pull request by jairomoldes**
I hope this fixes issue-22

### CODE REVIEW

File: src/boost/cpp/server/device_impl.cpp [hunk 1]
```diff
@@ -484,6 +484,16 @@ namespace PyDeviceImpl
...
     }
 
+    /* **********************************
+     * data ready event
+     * **********************************/
+    inline void push_data_ready_event(Tango::DeviceImpl &self, const str &name,
+                                      long ctr)
+    {
+        SAFE_PUSH(self, attr, name)
+        self.push_data_ready_event(__att_name, ctr); //__att_name from SAFE_PUSH
```
Comment (vxgmichel) [inline src/boost/cpp/server/device_impl.cpp]: I noticed that `push_archive_event` uses `attr.fire_archive_event()`. Is there any reason why you're not doing something similar, e.g `attr.fire_data_ready_event()`?

**Comment by vxgmichel** [comment]
Has it been tested against the example scenario in issue #22?

File: src/boost/cpp/server/device_impl.cpp (hunk 1 above)
Comment (jairomoldes) [inline src/boost/cpp/server/device_impl.cpp]: It looks like Attribute cpp class only provides these methods:

- Attribute::fire_archive_event
- Attribute::fire_change_event
- Attribute::fire_event
- Attribute::fire_error_periodic_event

I can't see any "fire_data_ready_event" nor anything similar that could be used. Any suggestion?

**Comment by jairomoldes** [comment]
Yes, I tested the solution using the sample code in https://github.com/tango-cs/pytango/issues/22
I left it running ~24 hours and got no error.

File: src/boost/cpp/server/device_impl.cpp (hunk 1 above)
Comment (vxgmichel) [inline src/boost/cpp/server/device_impl.cpp]: OK, I was just curious about it.

**Comment by vxgmichel** [comment]
Great! Then it should be safe to merge.

**Comment by vxgmichel** [comment]
Thanks!

**Pull request merged**